src/frontend/src/optimizer/property/monotonicity.rs

```diff
@@ -271,3 +292,53 @@ impl MonotonicityAnalyzer {
         Inherent(Unknown)
     }
 }
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, Hash)]
+pub struct MonotonicityMap(BTreeMap<usize, Monotonicity>);
+
+impl MonotonicityMap {
+    pub fn new() -> Self {
+        MonotonicityMap(BTreeMap::new())
+    }
+
+    pub fn insert(&mut self, idx: usize, monotonicity: Monotonicity) {
+        if monotonicity != Monotonicity::Unknown {
+            self.0.insert(idx, monotonicity);
+        }
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = (usize, Monotonicity)> + '_ {
+        self.0
+            .iter()
+            .map(|(idx, monotonicity)| (*idx, *monotonicity))
+    }
+}
+
+impl Index<usize> for MonotonicityMap {
+    type Output = Monotonicity;
+
+    fn index(&self, idx: usize) -> &Self::Output {
+        self.0.get(&idx).unwrap_or(&Monotonicity::Unknown)
+    }
+}
+
+impl IndexMut<usize> for MonotonicityMap {
+    fn index_mut(&mut self, idx: usize) -> &mut Self::Output {
+        self.0.entry(idx).or_insert(Monotonicity::Unknown)
+    }
+}
```

Comment:
If you want to maintain the invariant of not containing `Unknown`, I think we should not implement `IndexMut`, and stick to `insert`, because you will be able to `m[i]=Unknown`.